src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -484,6 +484,11 @@ def is_saveable_module(name, value):
             sub_model = getattr(self, pipeline_component_name)
             model_cls = sub_model.__class__
 
+            # Dynamo wraps the original mode and changes the class.
+            # Is there a principled way to obtain the original class?
+            if "_dynamo.eval_frame" in str(model_cls):
```

Comment:
Think that's a bit cleaner
```suggestion
            if is_torch_version(">=", "2.0.0") and isinstance(sub_model, torch._dynamo.eval_frame.OptimizedModule):
```

Reply:
@pcuenca quick question - why don't we go for this API? Think `isinstance` is a better check here than the string checking

Reply:
Because `torch._dynamo` is not available in all platforms, just on cuda builds.

Reply:
Think that's a bit cleaner
```suggestion
            if is_torch_version(">=", "2.0.0") and hasattr(torch, "_dynamo") and isinstance(sub_model, torch._dynamo.eval_frame.OptimizedModule):
```

Reply:
I see! Tbh, both ways are fine by me - I do prefer to avoid silent errors and generally don't like string "in" testing for classes as this is a very brittle test in my opinion. 

Would prefer a helper function here, but also ok to leave as is as it's an edge case.